packages/client/src/v2-events/components/forms/FormFieldGenerator.tsx

```diff
@@ -391,6 +408,25 @@ class FormSectionComponent extends React.Component<AllProps> {
   }
 }
 
+const FIELD_SEPARATOR = '____'
+function unflatten<T>(data: Record<string, T>) {
+  return Object.fromEntries(
+    Object.entries(data).map(([key, value]) => [
+      key.replaceAll('.', FIELD_SEPARATOR),
+      value
+    ])
+  )
+}
+
+function flatten<T>(data: Record<string, T>) {
+  return Object.fromEntries(
+    Object.entries(data).map(([key, value]) => [
+      key.replaceAll(FIELD_SEPARATOR, '.'),
+      value
+    ])
+  )
+}
```

Comment:
Let's document this very carefully why we do this, it's so annoying :(

Reply:
Yup :D